dodo/search.py

```diff
@@ -42,6 +42,7 @@ class SearchModel(QAbstractItemModel):
     def __init__(self, q: str) -> None:
         super().__init__()
         self.q = q
+        self._count = 0
```

Comment:
Maybe `-1` or `None` given that `0` can be a valid count as well?

Reply:
In the new commits, there's some error handling for the wider update that just keeps the previous data around when something goes wrong. Under that rationale, I prefer keeping the last known valid count rather than use a marker error. It makes sense as presumably, the error is most likely a transient one anyway.